core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/handler/BaseAsyncClientHandler.java

```diff
@@ -65,7 +66,9 @@ protected BaseAsyncClientHandler(SdkClientConfiguration clientConfiguration,
         super(clientConfiguration);
         this.clientConfiguration = clientConfiguration;
         this.client = client;
-        this.crc32Validator = response -> Crc32Validation.validate(isCalculateCrc32FromCompressedData(), response);
+        this.crc32Validator =
+            response -> Crc32Validation
+                .validate(clientConfiguration.option(SdkClientOption.CRC32_FROM_COMPRESSED_DATA_ENABLED), response);
```

Comment:
I actually like the original version. IMO chained method calls is a bit hard to read.